pkg/xen-tools/initrd/chroot2.c

```diff
@@ -27,13 +28,42 @@ static int childFunc(void *args)
 {
     clone_args *parsed_args = (clone_args *)args;
 
-    chroot(parsed_args->chroot);
-    chdir(parsed_args->workdir);
+    if (chroot(parsed_args->chroot) != 0) {
+        perror("chroot() failed:");
+        return -1;
+    }
+
+    if (chdir(parsed_args->workdir) != 0) {
+        perror("chdir() failed:");
+        return -1;
+    }
+
+    if (mount("proc", "/proc", "proc", 0, NULL) != 0) {
```

Comment:
I think this mount is some leftover from former implementation... we are already mounting the /proc in the initrd script, as well as the other needed filesystems (/dev, /sys) before call the chroot. I think we can safely remove this mount and just expect that /proc will be ready like the others... as the POSIX chroot does, it won't mount any of these filesystems by himself...

Reply:
clone() is called with CLONE_NEWPID so AFAIR mounting it allows the launched process and all the subsequent childs have their own (limited) view of the /proc.

Reply:
but I think we need to create a new mnt namespace for the child, because this mount affects the parent process too.

I will send another PR for that.